modules/tensor_mechanics/include/userobjects/GeneralizedPlaneStrainUserObject.h

```diff
@@ -26,8 +26,8 @@ class GeneralizedPlaneStrainUserObject : public ElementUserObject
   void execute() override;
   void threadJoin(const UserObject & uo) override;
   void finalize() override;
-  virtual Real returnResidual() const;
-  virtual Real returnJacobian() const;
+  virtual Real returnResidual(unsigned int component = 0) const;
+  virtual Real returnJacobian(unsigned int component = 0) const;
```

Comment:
Per our discussion, let's rename `component` to `scalar_variable_index` to avoid confusion with other usages of `component`.